src/conditions.rs

```diff
@@ -42,24 +42,28 @@ impl Conditions {
     /// - compose output structure
     /// - convert output to JSON and return
     pub fn fetch(&self) -> eyre::Result<String> {
-        let provider = match &self.config.weatherapi_token {
-            Some(token) => {
-                crate::weather::Provider::WeatherAPI(token.to_string())
-            }
-            None => crate::weather::Provider::OpenMeteo,
-        };
+        let mut providers = Vec::new();
+
+        if let Some(api_key) = &self.config.weatherapi_token {
+            providers.push(Provider::WeatherAPI(api_key.to_string()));
+        }
+
+        providers.push(Provider::OpenMeteo);
 
         let location = self.config.get_location()?;
 
-        let mut conditions = crate::weather::CurrentConditions::get(
-            provider.clone(),
+        let mut conditions = CurrentConditions::get(
+            providers,
             self.config.unit,
             &location.latitude,
             &location.longitude,
         )?;
 
-        conditions
-            .set_icon(conditions.time_of_day.icon(provider, conditions.code));
+        conditions.set_icon(
+            conditions
+                .time_of_day
+                .icon(conditions.provider.clone(), conditions.code),
+        );
```

Comment:
Consider allowing the `WeatherProvider` to set the icon. Perhaps by passing it down?